curifactory/staging.py

```diff
@@ -560,11 +634,22 @@ def wrapper(record: Record, records: list[Record] = None, **kwargs):
                     f"Stage '{name}' - the number of cachers does not match the number of outputs to cache"
                 )
 
-            # at this point we've grabbed all information we would need if we're
-            # just mapping out the stages, so return at this point.
-            if record.manager.map_mode:
-                record.manager.stage_active = False
-                return record
+            # similar to how inputs are checked on the state for a regular stage, we use
+            # `expected_state` to (softly) check each input record for the requested state
+            # variables. We don't actually load/pass anything differently to the function,
+            # but this is useful from a documentation and debugging standpoint, and is required
+            # for DAG computation to work as expected.
+            for expected_input in expected_state:
+                for prev_record in records:
+                    if expected_input not in prev_record.state_artifact_reps:
+                        logging.warning(
+                            "Expected state variable '%s' not found in %s"
+                            % (expected_input, prev_record.get_reference_name())
+                        )
+                    else:
+                        record.stage_inputs[-1].append(

```

Comment:
I don't fully understand why it is always appending to the last `stage_inputs`.

Reply:
`stage_inputs` (and `stage_outputs`) is technically a list of lists of strings, each stage has its own list of strings, so we're always recording this input on the last stage (the current one)